### ISSUE TITLE
Support inplace changes for `hcledit set` 

### ISSUE THREAD

**Issue by 3h4x** [minamijoyo/hcledit#4]
Because redirecting stdout directly to file that we want to modify will remove the content
`cat versions.tf | hcledit attribute set terraform.required_version '"=> 0.12.26"' > versions.tf`
👎 

 I'm forced to do some cp/mv tricks:
`cat versions.tf | hcledit attribute set terraform.required_version '"=> 0.12.26"' > tmp.tf && mv tmp.tf versions.tf`

Would love to see inplace feature like in `sed`:
>      -i extension
>             Edit files in-place, saving backups with the specified extension.  If a
>             zero-length extension is given, no backup will be saved.  It is not rec-
>             ommended to give a zero-length extension when in-place editing files, as
>             you risk corruption or partial content in situations where disk space is
>             exhausted, etc.

**Comment by minamijoyo** [comment]
@3h4x Thank you for opening the issue.
Yes, I agree with you. The inplace update option sounds good to me.

Generally speaking, redirecting output to an input file will remove contents. Yes, it's true for most of unix-style commands. However, strictly speaking, the hcledit actually reads all input stream before writing any contents to output stream. I know this violates the unix philosophy. This is just because the hcl parser which we depends on is currently only allowed to pass byte arrays, not stream.
So as a workaround, you can redirect outputs to the same file if no errors.
But if an error occurs the contents will be lost. Oops, that's apparently unexpected behavior.
Also from a UX perspective, if you want to process all the files in a directory at once, it's probably more convenient to simply ignore them if there is no matching attribute for the address. Simply redirecting the file is inconvenient for such use-case.
I'll improve it to be more suitable for the real world use-case when I have time. Thank you for your feedback!

FYI: If your concern is only updating terraform version, the tfupdate is specialized for it. Check it out if you missed it.
https://github.com/minamijoyo/tfupdate

**Comment by minamijoyo** [comment]
@3h4x  Fixed in v0.2.0. For details, see #25

```
[hcledit@master|✔]$ cat tmp/attr.hcl
resource "foo" "bar" {
  attr1 = "val1"
  nested {
    attr2 = "val2"
  }
}

[hcledit@master|✔]$ hcledit attribute set -f tmp/attr.hcl -u resource.foo.bar.attr1 '"val3"'
[hcledit@master|✔]$ cat tmp/attr.hcl
resource "foo" "bar" {
  attr1 = "val3"
  nested {
    attr2 = "val2"
  }
}
```
